22009 (#2519)
OpenLiberty/blogs: posts/2022-08-30-password-utilities-22009.adoc

---
layout: post

> title: "Stand-alone user registries are no longer forcefully federated with the new Password Utilities 1.1 feature now in Open Liberty 22.0.0.9"

Reviewer: ```suggestion
title: "Use the Password Utilities feature without forcefully federating stand alone user registries in Open Liberty 22.0.0.9"
```

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/reecenana
author_github: https://github.com/reecenana

> seo-title: Stand-alone user registries are no longer forcefully federated with the new Password Utilities 1.1 feature now in Open Liberty 22.0.0.9 - OpenLiberty.io

Reviewer: ```suggestion
seo-title: Use the Password Utilities feature without forcefully federating stand alone user registries in Open Liberty 22.0.0.9 - OpenLiberty.io
```

> seo-description: Part of the core edition, password Utilities 1.1 has been introduced in order to no longer start the federatedRegistry-1.0 feature or the JCA / connectors features.

Reviewer: ```suggestion
seo-description: The Password Utilities 1.1 feature does not start the Federated User Registry feature or the Jakarta Connectors feature. When you use this version of the feature, stand-alone user registries are not forcefully federated.
```

> blog_description: Part of the core edition, password Utilities 1.1 has been introduced in order to no longer start the federatedRegistry-1.0 feature or the JCA / connectors features.

Reviewer: ```suggestion
blog_description: The Password Utilities 1.1 feature does not start the Federated User Registry feature or the Jakarta Connectors feature. When you use this version of the feature, stand-alone user registries are not forcefully federated.
```

open-graph-image: https://openliberty.io/img/twitter_card.jpg
---

> = Stand-alone user registries are no longer forcefully federated with the new Password Utilities 1.1 feature now in Open Liberty 22.0.0.9

Reviewer: ```suggestion
= Use the Password Utilities feature without forcefully federating stand alone user registries in Open Liberty 22.0.0.9
```

Reece Nana <https://github.com/reecenana>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> Open Liberty 22.0.0.9 has arrived and brings thew Password Utilities 1.1 which no longer forcefully federates stand-alone user registries. Along with the new feature, this release also includes several notable bug fixes.

Reviewer: ```suggestion
Open Liberty 22.0.0.9 has arrived and brings the Password Utilities 1.1 feature, which does not forcefully federate stand-alone user registries. Along with the new feature, this release also includes several notable bug fixes.
```

In link:{url-about}[Open Liberty] 22.0.0.9:

* <<password, Stand-alone user registries no longer forcefully federated (Password Utilities 1.1)>>
* <<CVEs, Security Vulnerability (CVE) Fixes>>
* <<bugs, Notable bug fixes>>

[#run]
== Run your apps using 22.0.0.9

If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:

[source,xml]
----
<dependency>
    <groupId>io.openliberty</groupId>
    <artifactId>openliberty-runtime</artifactId>
    <version>22.0.0.9</version>
    <type>zip</type>
</dependency>
----

Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:

[source,gradle]
----
dependencies {
    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[22.0.0.9,]'
}
----

Or if you're using Docker:

[source]
----
FROM open-liberty
----

Or take a look at our link:{url-prefix}/downloads/[Downloads page].

[link=https://stackoverflow.com/tags/open-liberty]
image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]

// https://github.com/OpenLiberty/open-liberty/issues/21962
[#password]
== Stand-alone user registries no longer forcefully federated (Password Utilities 1.1)

> The new `passwordUtilities-1.1` feature no longer starts the `federatedRegistry-1.0` feature or the Jakarta Connectors feature.

Reviewer: ```suggestion
A new version of the Password Utilities feature, passwordUtilities-1.1, is available. This version of the feature does not start the Federated User Registry feature or the Jakarta Connectors feature. When you use this version of the feature, stand-alone user registries are not forcefully federated, which sometimes results in slightly different behavior than the previous version. The previous feature version, passwordUtilities-1.0, starts the Federated User Registry and Jakarta Connectors features by default. 
```

> Stand-alone user registries are no longer forcefully federated, which sometimes results in slightly different behavior.

Reviewer: ```suggestion
```

To enable Password Utilities 1.1, add the `passwordUtilities-1.1` feature to the list of features in your server.xml file:

[source, xml]
----
<featureManager>
    <feature>passwordUtilities-1.1</feature>
</featureManager>
----

> For more information about the Password Utilities, check out the link:{url-prefix}/docs/latest/reference/feature/passwordUtilities.html[feature documentation].

Reviewer: ```suggestion
For more information about the Password Utilities feature, check out the link:{url-prefix}/docs/latest/reference/feature/passwordUtilities.html[feature documentation].
```

[#CVEs]
== Security vulnerability (CVE) fixes in this release

There are no security vulnerability fixes for this release. 
For a list of past security vulnerability fixes, reference the link:{url-prefix}/docs/latest/security-vulnerabilities.html[Security vulnerability (CVE) list].

[#bugs]
== Notable bugs fixed in this release

We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22009+label%3A%22release+bug%22+[full list of bugs fixed in 22.0.0.9].

* link:https://github.com/OpenLiberty/open-liberty/issues/22040[Invalid character warning for colon in WorkQueueManagerImplMBeanWrapper objectName]
+

> A colon character in the Apache CXF WorkQueueManagerImplMBeanWrapper class's objectName creates a warning and a `javax.management.MalformedObjectNameException` caused by the invalid `:` character when `javax.management.ObjectName.construct` is called. This issue occurs in Open Liberty 22.0.0.8 when it did not appear in 22.0.0.7.

Reviewer: ```suggestion
A colon character ( `:` ) in the Apache CXF WorkQueueManagerImplMBeanWrapper class's objectName creates a warning and a `javax.management.MalformedObjectNameException` caused by the invalid `:` character when `javax.management.ObjectName.construct` is called. This issue occurs in Open Liberty 22.0.0.8, though it did not appear in 22.0.0.7.
```

+

> A fix to this issue was made where by the removal of an invalid `:` character from URI base in the bus ID resulted in no additional warning messages. More information can be found in link:https://github.com/OpenLiberty/open-liberty/pull/22069[#22069]

Reviewer: ```suggestion
A fix to this issue was made where by the removal of an invalid `:` character from URI base in the bus ID and resulted in no additional warning messages. More information can be found in link:https://github.com/OpenLiberty/open-liberty/pull/22069[#22069]
```

* link:https://github.com/OpenLiberty/open-liberty/issues/22012[CXF property cxf.ignore.unsupported.policy is not processed correctly in Liberty 22.0.0.8]
+
In Liberty 22.0.0.8, when the JVM system property `cxf.ignore.unsupported.policy` is set to true, some unsupported policy assertions in the WSDL file are not ignored as expected.
+

> This has since been resolved to now process link:https://github.com/OpenLiberty/open-liberty/pull/22013[`cxf.ignore.unsupported.policy` correctly].

Reviewer: ```suggestion
This has since been resolved to now link:https://github.com/OpenLiberty/open-liberty/pull/22013[process the `cxf.ignore.unsupported.policy` correctly].
```

* link:https://github.com/OpenLiberty/open-liberty/issues/21973[Expiration fields are not compared in an LTPA Token]
+

> When there are two expiration times present in an LTPA token, the times should be compared to be within an acceptable difference. 

Reviewer: ```suggestion
When two expiration times are present in an LTPA token, the times must be compared to confirm whether they are within an acceptable difference. 
```

+
The issue is fixed via link:https://github.com/OpenLiberty/open-liberty/pull/21883[#21883] and 
Open Liberty will now reject LTPA tokens when the two expiration times in the token are not within an acceptable difference by throwing an invalidTokenException.

* link:https://github.com/OpenLiberty/open-liberty/issues/21955[Liberty does not provide exported packages for java.* packages at runtime in the OSGi framework instance]
+

> Starting with OSGi R7 Core specification, bundles are allowed to import (using `Import-Package` header) packages from `java.*` (e.g. `java.io`).  When installing a liberty feature that includes a bundle that imports a `java.*` package the bundle fails to resolve because Liberty does not configure the OSGi framework (Equinox in this case) to export the `java.*` packages available in the running JVM.

Reviewer: ```suggestion
Starting with OSGi R7 Core specification, bundles are allowed to import (using `Import-Package` header) packages from `java.*` (e.g. `java.io`).  When installing a Liberty feature that includes a bundle that imports a `java.*` package, the bundle fails to resolve because Liberty does not configure the OSGi framework (Equinox in this case) to export the `java.*` packages available in the running JVM.
```

+
When a method times out at the same time as it completes, an FFDC may be logged. Aside from the FFDC being logged, there are no other symptoms.
+
There has been a fix to timeout cancellation race condition which resolves:
+
A lock shared between `stop()` and `timeout()`, but if the timeout
has fired, the timeout future is not complete at the point where
`timeout()` releases the lock. This gives `stop()` a chance to run and
attempt to cancel the future. Doing this results in an
`InterruptedException` which causes and FFDC to be emitted by the policy
executor.
+
By checking the value of timeout, it avoids trying to cancel the
timeout task if the timeout has already fired and interrupted our
thread. There has also been a package added to the FT trace group.
+ 
Both the fix to the timeout cancellation race condition and adding the package to trace group has resolved the issue of link:https://github.com/OpenLiberty/open-liberty/pull/21936[MP Fault Tolerance 1.x]. 

* link:https://github.com/OpenLiberty/open-liberty/issues/21880[OpenAPI 2.0+ throws error at startup]
+

> After enabling mpOpenApi-2.0 or mpOpenApi-3.0. at startup, an error which can be found 

Reviewer: ```suggestion
After enabling mpOpenApi-2.0 or mpOpenApi-3.0. at startup, link:https://github.com/OpenLiberty/open-liberty/issues/21880[this error] might occur because the MP OpenAPI implementation tries to get a MicroProfile Config object before the MicroProfile Config feature has initialized.
```

> link:https://github.com/OpenLiberty/open-liberty/issues/21880[here] may occur because the MP OpenAPI implementation tries to get a MicroProfile Config object before the MicroProfile Config feature has initialized.

Reviewer: ```suggestion
```

+
This issue was fixed by only calling `Config.getConfig()` when needed, rather than when

> the component starts to avoid occasional cases where the config component hasn't initialized yet. The fixes around found in link:https://github.com/OpenLiberty/open-liberty/pull/21881[#21881]

Reviewer: ```suggestion
the component starts to avoid occasional cases where the config component hasn't initialized yet. The fixes are  in link:https://github.com/OpenLiberty/open-liberty/pull/21881[#21881]
```
